File: Number Operation/prg.cs
using System;

class Program {
    static void Main() {
        int a = int.Parse(Console.ReadLine()), b = int.Parse(Console.ReadLine());

        Console.WriteLine("AND: " + (a & b));
        Console.WriteLine("OR: " + (a | b));
        Console.WriteLine("XOR: " + (a ^ b));
        Console.WriteLine("NOT A: " + (~a));
        Console.WriteLine("Left Shift: " + (a << 1));
        Console.WriteLine("Right Shift: " + (a >> 1));

        bool x = bool.Parse(Console.ReadLine()), y = bool.Parse(Console.ReadLine());
        Console.WriteLine("Logical AND: " + (x && y));
        Console.WriteLine("Logical OR: " + (x || y));
        Console.WriteLine("Logical NOT X: " + (!x));
    }
}

File: Number Operation/prg.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let values: Vec<i32> = input.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();
    
    let (a, b) = (values[0], values[1]);

    println!("AND: {}", a & b);
    println!("OR: {}", a | b);
    println!("XOR: {}", a ^ b);
    println!("NOT A: {}", !a);
    println!("Left Shift: {}", a << 1);
    println!("Right Shift: {}", a >> 1);

    let mut bool_input = String::new();
    io::stdin().read_line(&mut bool_input).unwrap();
    let bools: Vec<bool> = bool_input.trim().split_whitespace().map(|x| x.parse().unwrap()).collect();

    println!("Logical AND: {}", bools[0] && bools[1]);
    println!("Logical OR: {}", bools[0] || bools[1]);
    println!("Logical NOT X: {}", !bools[0]);
}
